chartfx-chart/src/main/java/de/gsi/chart/plugins/TableViewer.java

```diff
@@ -491,34 +507,90 @@ public int size() {
          * @author akrimm
          */
         protected class DataSetTableColumn extends TableColumn<DataSetsRow, Double> {
-            private final DataSet ds;
+            private DataSet ds;
             private final ColumnType type;
 
             /**
              * Creates a TableColumn with the text set to the provided string, with default comparator. The cell factory
              * and onEditCommit implementation facilitate editing of the DataSet column identified by the ds and type
              * Parameter
              * 
-             * @param text The string to show when the TableColumn is placed within the TableView
-             * @param dataSet The dataset containing the column
              * @param type The field of the data to be shown
              */
-            public DataSetTableColumn(final String text, final DataSet dataSet, final ColumnType type) {
-                super(text);
-                this.ds = dataSet;
+            public DataSetTableColumn(final ColumnType type) {
+                super("");
+                this.setSortable(false);
+                this.setReorderable(false);
+                this.ds = null;
                 this.type = type;
                 this.setCellValueFactory(dataSetsRowFeature -> new ReadOnlyObjectWrapper<>(
                         dataSetsRowFeature.getValue().getValue(ds, type)));
 
-                if (editable) {
-                    updateEditableState();
-                }
+                setVisible(false);
             }
 
             public double getValue(final int row) {
                 return dsModel.getValue(row, ds, type);
             }
 
+            public void update(final DataSet newDataSet) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'update(DataSet)' has an NPath complexity of 11680](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=4834981)